File: src/main/java/io/github/sd3v/mflashcardsbe/repository/FlashcardRepositoryImpl.java
package io.github.sd3v.mflashcardsbe.repository;

import io.github.sd3v.mflashcardsbe.repository.entity.FlashcardEntity;
import java.util.List;
import java.util.Optional;
import org.springframework.stereotype.Repository;

@Repository
public class FlashcardRepositoryImpl implements FlashcardRepository {
  @Override
  public Optional<Long> findById(Integer id) {
    return Optional.empty();
  }

  @Override
  public List<FlashcardEntity> findAllByDeckId(Integer id) {
    return null;
  }
}

File: src/main/kotlin/io/github/sd3v/mflashcardsbe/repository/FlashcardRepositoryImpl.kt
package io.github.sd3v.mflashcardsbe.repository

import io.github.sd3v.mflashcardsbe.jooq.tables.Flashcard.FLASHCARD
import io.github.sd3v.mflashcardsbe.jooq.tables.pojos.Flashcard
import io.github.sd3v.mflashcardsbe.repository.entity.FlashcardEntity
import org.jooq.DSLContext
import org.springframework.stereotype.Repository

@Repository
class FlashcardRepositoryImpl(val create: DSLContext) : FlashcardRepository {

    override fun findById(id: Long): FlashcardEntity? =
        create.select().from(FLASHCARD)
            .where(FLASHCARD.ID.equal(id))
            .fetchOne()
            ?.into(Flashcard::class.java)
            ?.let { FlashcardEntity(it.id, it.front, it.back, it.deckId) }

    override fun findAllByDeckId(deckId: Long): List<FlashcardEntity> =
        create.select().from(FLASHCARD)
            .where(FLASHCARD.DECK_ID.equal(deckId))
            .fetch()
            .into(Flashcard::class.java)
            .map { FlashcardEntity(it.id, it.front, it.back, it.deckId) }
}
